File: oscexchange/src/main/java/ac/at/fhstp/digitech/oscexchange/OSCParsedReceiveRequest.java
package ac.at.fhstp.digitech.oscexchange;

final class OSCParsedReceiveRequest<T> extends OSCReceiveRequest {

    public final OSCArgsParser<T> parser;
    public final OSCParsedReceiveListener<T> listener;


    OSCParsedReceiveRequest(OSCAddress address, OSCArgsValidator validator, OSCArgsParser<T> parser, OSCParsedReceiveListener<T> listener) {
        super(address, validator);
        this.parser = parser;
        this.listener = listener;
    }

}

File: oscexchange/src/main/java/ac/at/fhstp/digitech/oscexchange/OSCParsedReceiveRequest.kt
package ac.at.fhstp.digitech.oscexchange

internal class OSCParsedReceiveRequest<T>(
    address: OSCAddress,
    validator: OSCArgsValidator?,
    val parser: OSCArgsParser<T>,
    val listener: OSCParsedReceiveListener<T>?
) : OSCReceiveRequest(address, validator)